gossipsub.go

```diff
@@ -103,9 +182,19 @@ func (gs *GossipSubRouter) Protocols() []protocol.ID {
 func (gs *GossipSubRouter) Attach(p *PubSub) {
 	gs.p = p
 	gs.tracer = p.tracer
+
+	// start the scoring, if any
+	if gs.score != nil {
```

Comment:
Could we invert this and turn it on by default with some off-the-shelf reasonable values and make the consumer explicitly disable it if needed?

Reply:
I can just move the nil check inside the `Start` implementation so that it does not occlude the logic.

Reply:
I am not sure there we can come up with reasonable off-the-shelf defaults, let me think about it.

Reply:
for now I have moved this check inside `Start` so that it doesn't occlude the logic.

Reply:
So I have given it some more thought and I think we can make a gossipsub v1.1 constructor function that can fill in default parameters given the expected message rate and the application score function.